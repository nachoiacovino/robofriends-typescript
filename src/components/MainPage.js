import React from 'react'
import CardList from './CardList'
import SearchBox from './SearchBox'
import Scroll from './Scroll'
import Header from './Header'
import CounterButton from './CounterButton'

const MainPage = ({ robots, onSearchChange }) => {
    return (
        <div className='tc'>
            <Header />
            <CounterButton />
            <SearchBox onSearchChange={onSearchChange}/>
            <Scroll>
                <CardList robots={robots} />
            </Scroll>
        </div>
    )
}

export default MainPage
