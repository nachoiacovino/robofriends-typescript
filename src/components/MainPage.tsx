import React from "react";
import CardList from "./CardList";
import SearchBox from "./SearchBox";
import Scroll from "./Scroll";
import Header from "./Header";
import CounterButton from "./CounterButton";

interface IRobot {
  name: string;
  id: number;
  email: string;
}

interface ISearchBox {
  onSearchChange(e: React.SyntheticEvent<HTMLInputElement>): void;
}

const MainPage = ({
  robots,
  onSearchChange
}: {
  robots: Array<IRobot>;
  onSearchChange: ISearchBox;
}) => {
  return (
    <div className="tc">
      <Header />
      <CounterButton />
      <SearchBox onSearchChange={onSearchChange} />
      <Scroll>
        <CardList robots={robots} />
      </Scroll>
    </div>
  );
};

export default MainPage;
