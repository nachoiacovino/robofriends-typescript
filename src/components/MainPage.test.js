import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import MainPage from './MainPage'

it("expect to render MainPage component", () => {
    const mockStore = {
        robots: [],
        searchField: "",
        isPending: true,
        error: ""
    }

    expect(toJson(shallow(<MainPage />))).toMatchSnapshot()
})