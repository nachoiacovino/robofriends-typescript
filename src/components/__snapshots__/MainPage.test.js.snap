// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render MainPage component 1`] = `
<div
  className="tc"
>
  <Memo() />
  <CounterButton />
  <SearchBox />
  <Scroll>
    <CardList />
  </Scroll>
</div>
`;
